compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -403,23 +403,40 @@ impl<'f> LoopInvariantContext<'f> {
             }
             Binary(binary) => self.can_evaluate_binary_op(binary),
             Constrain(..) | ConstrainNotEqual(..) | RangeCheck { .. } => {
-                // These instructions should not be hoisted if we know the loop will never be executed (an upper bound or zero or equal loop bounds)
-                // or we are unsure if the loop will ever be executed (dynamic loop bounds).
-                // If the instruction were to be hoisted out of a loop that never executes it could potentially cause the program to fail when it is not meant to fail.
-                let bounds = self.current_induction_variables.values().next().copied();
-                let does_loop_body_execute = bounds
-                    .and_then(|(lower_bound, upper_bound)| {
-                        upper_bound.reduce(lower_bound, |u, l| u > l, |u, l| u > l)
-                    })
-                    .unwrap_or(false);
-                // If we know the loop will be executed these instructions can still only be hoisted if the instructions
-                // are in a non control dependent block.
-                does_loop_body_execute && !self.current_block_control_dependent
+                // If we know the loop will be executed we can still only hoist if we are in a non control dependent block.
+                self.does_loop_body_execute() && !self.current_block_control_dependent
```

Comment:
We could swap the elements in `&&` so that `does_loop_body_execute()` doesn't need to execute if `self.current_block_control_dependent` is false

Reply:
done